### ISSUE TITLE
"Could not resolve host" - even when using IP addresses, even on same LAN

### ISSUE THREAD

**Issue by aleqx** [ethereum-mining/ethminer#624]
Ok, this is a tad ridiculous. I'm not necessarily  blaming ethminer, I know it's possible that the problem stems from boost, and I know there is an issue about failing to resolve DNS (https://github.com/ethereum-mining/ethminer/issues/259), but now this happens even for IP addresses.

I connected to a local machine sitting on the **same LAN**, connected via a switch, and address it **using an IP address** ... no DNS name resolution required. 

`./ethminer -U -S 192.168.10.9:5555 -O 0xblabla..bla.testnode:x`

The miner's IP is 192.168.10.212. This is what happens at launch virtually every time (very rarely it connects straight away)

```
  ℹ  02:19:14|ethminer  Connecting to stratum server 192.168.10.9:5555
Could not resolve host 192.168.10.9:5555, Operation canceled  ℹ  02:19:14|ethminer  Reconnecting in 3 seconds...
  ℹ  02:19:17|ethminer  Connecting to stratum server 192.168.10.9:5555
Could not resolve host 192.168.10.9:5555, Operation canceled  ℹ  02:19:17|ethminer  Reconnecting in 3 seconds...
  ℹ  02:19:20|ethminer  Connecting to stratum server 192.168.10.9:5555
Could not resolve host 192.168.10.9:5555, Operation canceled  ℹ  02:19:20|ethminer  Reconnecting in 3 seconds...
[...]
```
Sometimes the above repeats for **several minutes** on end ... takes a toll on profits when automated restarts are needed.

What host resolution? There's no DNS here. Obviously the error is misleading. If the issue stems from boost, then could it be worked around in ethminer first before calling boost?

**Comment by MasonLyons** [comment]
Can confirm. I'm also getting this error after going to the newest stable version.

**Comment by aleqx** [comment]
No love for this?

**Comment by stralex7** [comment]
Same here 0.13

**Comment by aleqx** [comment]
Sweet! I moved on from Ethereum mining for now, but if this was indeed fixed then it's sweet nonetheless!

**Comment by wargun02** [comment]
Can confirm this issue. I'm running on Ubuntu 16, with 12 1060 GPUs.

Sometimes, it is fine by looking at the log, but when I checked on the pool stats, it has no submission, did a restart, this message came about: "Could not resolve host xxxpool.org:19999, Operation canceled  ℹ  09:40:22|ethminer  Reconnecting in 3 seconds...

Tried to change multiple pool stratum url, but it just doesn't want to connect.
